src/subdomains/user/application/services/lightning-wallet.service.ts

```diff
@@ -0,0 +1,166 @@
+  private readonly logger = new LightningLogger(LightningWalletService);
+
+  private readonly client: LightningClient;
+
+  constructor(
+    lightningService: LightningService,
+    private readonly lightningTransactionService: LightningTransactionService,
+    private readonly userTransactionRepository: UserTransactionRepository,
+    private readonly lightingWalletRepository: LightingWalletRepository,
+  ) {
+    this.client = lightningService.getDefaultClient();
+  }
+
+  @Cron(CronExpression.EVERY_HOUR)
+  @Lock()
+  async processUpdateLightningWalletBalances(): Promise<void> {
+    if (Config.processDisabled(Process.UPDATE_WALLET_BALANCE)) return;
+
+    try {
+      let lightningWalletEntities = await this.lightingWalletRepository.first(1000);
+
+      while (lightningWalletEntities.length) {
+        await this.doProcessUpdateLightningWalletBalances(lightningWalletEntities);
+        lightningWalletEntities = await this.lightingWalletRepository.next();
+      }
+    } catch (e) {
+      this.logger.error('Error while updating wallet balances', e);
+    }
+  }
+
+  private async doProcessUpdateLightningWalletBalances(
+    lightningWalletEntities: LightningWalletEntity[],
+  ): Promise<void> {
+    for (const lightningWalletEntity of lightningWalletEntities) {
+      const lnbitsWallet = await this.client.getLnBitsWallet(lightningWalletEntity.adminKey);
+
+      const lightningWalletEntityBalance = lightningWalletEntity.balance;
+      const lnbitsWalletBalance = LightningHelper.btcToSat(lnbitsWallet.balance);
+
+      if (lightningWalletEntityBalance !== lnbitsWalletBalance) {
+        await this.lightingWalletRepository.update(lightningWalletEntity.id, {
+          balance: lnbitsWalletBalance,
+        });
+      }
+    }
+  }
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  @Lock()
+  async syncLightningUserTransactions(): Promise<void> {
+    if (Config.processDisabled(Process.UPDATE_LIGHTNING_USER_TRANSACTION)) return;
+
+    const databaseUserTransactionEntities = await this.userTransactionRepository.getEntriesWithMaxCreationTimestamp();
+
+    if (0 === databaseUserTransactionEntities.length) {
+      await this.saveLightningUserTransactions();
+      return;
+    }
+
+    await this.saveLightningUserTransactions(databaseUserTransactionEntities);
+  }
+
+  private async saveLightningUserTransactions(
+    databaseUserTransactionEntities?: UserTransactionEntity[],
+  ): Promise<UserTransactionEntity[]> {
+    const newUserTransactionEntities: UserTransactionEntity[] = [];
+
+    const allUserWallets = await this.client.getUserWallets();
+
+    const startDate = databaseUserTransactionEntities
+      ? new Date(databaseUserTransactionEntities[0].creationTimestamp.getTime())
+      : new Date(0);
+
+    for (const wallet of allUserWallets) {
+      const userTransactionEntities = await this.getUserWalletTransactions(startDate, wallet);
+      newUserTransactionEntities.push(...userTransactionEntities);
+    }

```

Comment:
Could we miss some transactions due to using the same `startDate` for all wallets?